spec/struct.dd

```diff
@@ -566,6 +566,206 @@ $(GNAME Postblit):
+struct Z
+{
+    Y a;
+    this(this) {}
+}
+
+void main()
+{
+    // X has __postblit and __xpostblit (pointing to __postblit)
+    static assert(__traits(hasMember, X, "__postblit"));
+    static assert(__traits(hasMember, X, "__xpostblit"));
+
+    // Y does not have __postblit, but has __xpostblit (pointing to __fieldPostblit)
+    static assert(!__traits(hasMember, Y, "__postblit"));
+    static assert(__traits(hasMember, Y, "__xpostblit"));
+    // __fieldPostblit is not a member of the struct
+    static assert(!__traits(hasMember, Y, "__fieldPostblit"));
+
+    // Z has  __postblit and __xpostblit (pointing to __aggrPostblit)
+    static assert(__traits(hasMember, Z, "__postblit"));
+    static assert(__traits(hasMember, Z, "__xpostblit"));
+    // __aggrPostblit is not a member of the struct
+    static assert(!__traits(hasMember, Z, "__aggrPostblit"));
+}
+---
+)
+
+        $(P Neither of the above postblits is defined for structs that don't
+        define `this(this)` and don't have fields that transitively define it.
+        Example:)
+
+$(SPEC_RUNNABLE_EXAMPLE_COMPILE
+---
+struct X
+{}
+
+struct Y
+{
+    int a;
+    X b;
+}
+
+void main()
+{
+    static assert(!__traits(hasMember, X, "__postblit"));
+    static assert(!__traits(hasMember, X, "__xpostblit"));
+
+    static assert(!__traits(hasMember, Y, "__postblit"));
+    static assert(!__traits(hasMember, Y, "__xpostblit"));
+}
+---
+)
+
+        $(P Postblits are not overloadable. If two or more postblits are defined,
+        even if the signatures differ, the compiler assigns the
+        `__postblit` name to both and later issue a conflicting function
+        name error:)
+
+---
+struct X
+{
+    this(this) {}
+    this(this) const {} // error: function X.__postblit conflicts with function X.__postblit
+}
+---
+
+        $(P The following describes the behavior of the
+        qualified postblit definitions:)
+
+        $(OL
+        $(LI `const`. When a postblit is qualified with `const` as in
+        $(D this(this) const;) or $(D const this(this);) then the postblit
+        is succesfully called on mutable (unqualified), `const`,
+        and `immutable` objects, but the postblit cannot modify the object
+        because it regards it as `const`; hence `const` postblits are of
+        limited usefulness. Example:)
+
+$(SPEC_RUNNABLE_EXAMPLE_COMPILE
+---
+struct S
+{
+    int n;
+    this(this) const
+    {
+        import std.stdio : writeln;
+        writeln("postblit called");
+        //++n; // error: cannot modify this.n in `const` function
+    }
+}
+
+void main()
+{
+    S s1;
+    auto s2 = s1;
+    const S s3;
+    auto s4 = s3;
+    immutable S s5;
+    auto s6 = s5;
+}
+---
+)
+        $(LI `immutable`. When a postblit is qualified with `immutable`
+        as in $(D this(this) immutable) or $(D immutable this(this))
+        the code is ill-formed. The `immutable` postblit passes the
+        compilation phase but cannot be invoked. Example:)
```

Comment:
Version when this will be diagnosed? I think we can leave this text as is and add a mention once we merge the appropriate PR.